service/k8sclient/internal/kubectl/kubectl.go

```diff
@@ -283,3 +332,130 @@ func run(ctx context.Context, kubectlCmd []string, args []string, stdin interfac
 	l.Debug(errBuf.String())
 	return outBuf.Bytes(), err
 }
+
+var (
+	reservedProxyPortsMutex *sync.Mutex = new(sync.Mutex)
+	reservedProxyPorts      *sync.Map   = new(sync.Map)
+)
+
+func (k *KubeCtl) reserveProxyPort(port string) {
+	mutex, _ := reservedProxyPorts.LoadOrStore(port, new(sync.Mutex))
+	mutex.(*sync.Mutex).Lock()
+}
+
+func (k *KubeCtl) releaseProxyPort(port string) {
+	mutex, ok := reservedProxyPorts.Load(port)
+	if !ok {
+		return
+	}
+	mutex.(*sync.Mutex).Unlock()
```

Comment:
This implementation leaves mutex that protect certain port in the map for future use. I set range of ports to be [10000, 19999). Is this a good idea? @askomorokhov @BupycHuk

Reply:
It feels like decreasing the port range should be better. We can then reuse mutexes that were created. Would be 100 ports enough? Are we going to handle more than 100 concurrent calls to GetResources rpc?

Reply:
well, actually the range of ports should be a bit higher than number of expected concurrent users right?

Reply:
Implementation changed. outdated.